lockfile/src/golang.rs

```diff
@@ -66,4 +110,258 @@ mod tests {
             assert!(pkgs.contains(&expected_pkg));
         }
     }
+
+    #[test]
+    fn parse_go_mod() {
+        let pkgs = GoMod.parse(include_str!("../../tests/fixtures/go.mod")).unwrap();
+        assert_eq!(pkgs.len(), 5);
+
+        let expected_pkgs = [
+            Package {
+                name: "github.com/go-chi/chi/v5".into(),
+                version: PackageVersion::FirstParty("v5.0.12".into()),
+                package_type: PackageType::Golang,
+            },
+            Package {
+                name: "github.com/rs/zerolog".into(),
+                version: PackageVersion::FirstParty("v1.32.0".into()),
+                package_type: PackageType::Golang,
+            },
+            Package {
+                name: "github.com/mattn/go-colorable".into(),
+                version: PackageVersion::FirstParty("v0.1.13".into()),
+                package_type: PackageType::Golang,
+            },
+            Package {
+                name: "github.com/mattn/go-isatty".into(),
+                version: PackageVersion::FirstParty("v0.0.20".into()),
+                package_type: PackageType::Golang,
+            },
+            Package {
+                name: "golang.org/x/sys".into(),
+                version: PackageVersion::FirstParty("v0.12.0".into()),
+                package_type: PackageType::Golang,
+            },
+        ];
+
+        for expected_pkg in expected_pkgs {
+            assert!(pkgs.contains(&expected_pkg));
+        }
+    }
+
+    #[test]
+    fn parse_go_mod_unsupported() {
+        let go_mod_content = r#"
+            module cli/example
+            
+            go 1.14
+            
+            require (
+                github.com/go-chi/chi/v5 v5.0.12
+                github.com/rs/zerolog v1.32.0
+            )
+            
+            require (
+                github.com/mattn/go-colorable v0.1.13 // indirect
+                github.com/mattn/go-isatty v0.0.20 // indirect
+                golang.org/x/sys v0.12.0 // indirect
+            )
+        "#;
+
+        let error = GoMod.parse(go_mod_content).err().unwrap();
+        assert_eq!(error.to_string(), "Minimum supported go directive is 1.17")
+    }
+
+    #[test]
+    fn parse_go_mod_replace() {
+        let go_mod_content = r#"
+            module cli/example
+            
+            go 1.17
+
+            require (
+                example.com/othermodule v1.2.3
+                example.com/thismodule v1.2.3
+                example.com/thatmodule v1.2.3
+            )
+
+            replace example.com/thatmodule => ../thatmodule
+            replace example.com/thismodule v1.2.3 => example.com/newmodule v3.2.1
```

Comment:
Couldn't this be part of our sample fixture file? I like having a file that contains all kinds of edgecases since that makes it easier to test things.

Same goes for most of the other tests.

Reply:
Combined the tests and sample fixture except for the unsupported go directive test case.